### ISSUE TITLE
Queue-scheduler: Threads blocked - Artemis.sendMessage doesn't return

### ISSUE THREAD

**Issue by calohmn** [EnMasseProject/enmasse#742]
I've done a test with a 0.16 snapshot from 2018-01-16 (commit `5b32a98b`). I've added 13 addresses of type "queue" with plac "pooled-persisted"

The log output of the queue-scheduler container contains repeated warnings like this:
````
Jan 19, 2018 11:03:22 AM io.vertx.core.impl.BlockedThreadChecker
WARNING: Thread Thread[vert.x-worker-thread-8,5,main] has been blocked for 2371642 ms, time limit is 60000
io.vertx.core.VertxException: Thread blocked
	at io.enmasse.queue.scheduler.SchedulerState.addressesChanged(SchedulerState.java:46)
	at io.enmasse.queue.scheduler.QueueScheduler.lambda$addressesChanged$9(QueueScheduler.java:139)
	at io.enmasse.queue.scheduler.QueueScheduler$$Lambda$44/1584530467.run(Unknown Source)
	at io.enmasse.queue.scheduler.QueueScheduler.lambda$executeBlocking$10(QueueScheduler.java:145)
	at io.enmasse.queue.scheduler.QueueScheduler$$Lambda$45/1607717068.handle(Unknown Source)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:278)
	at io.vertx.core.impl.ContextImpl$$Lambda$47/1554489389.run(Unknown Source)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:80)
	at io.vertx.core.impl.TaskQueue$$Lambda$9/1285524499.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
````
It looks like there is a rising number of vertx worker threads blocked in that way, waiting for the `Artemis.sendMessage` invocation in this thread to finish:
````
"vert.x-worker-thread-16" #37 prio=5 os_prio=0 tid=0x00007f07902854f0 nid=0xa3 waiting on condition [0x00007f077b3ac000]
   java.lang.Thread.State: TIMED_WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000000fc6bbfd8> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
	at java.util.concurrent.LinkedBlockingDeque.pollFirst(LinkedBlockingDeque.java:522)
	at java.util.concurrent.LinkedBlockingDeque.poll(LinkedBlockingDeque.java:684)
	at io.enmasse.amqp.Artemis.sendMessage(Artemis.java:191)
	at io.enmasse.amqp.Artemis.doRequestResponse(Artemis.java:163)
	at io.enmasse.amqp.Artemis.doRequestResponse(Artemis.java:149)
	at io.enmasse.amqp.Artemis.doOperation(Artemis.java:132)
	at io.enmasse.amqp.Artemis.destroyConnectorService(Artemis.java:220)
	at io.enmasse.queue.scheduler.ArtemisAdapter.deleteQueue(ArtemisAdapter.java:55)
	at io.enmasse.queue.scheduler.SchedulerState.deleteQueue(SchedulerState.java:105)
	at io.enmasse.queue.scheduler.SchedulerState.deleteAddresses(SchedulerState.java:201)
	at io.enmasse.queue.scheduler.SchedulerState.groupUpdated(SchedulerState.java:64)
	- locked <0x00000000fc06da00> (a io.enmasse.queue.scheduler.SchedulerState)
	at io.enmasse.queue.scheduler.SchedulerState.addressesChanged(SchedulerState.java:51)
	- locked <0x00000000fc06da00> (a io.enmasse.queue.scheduler.SchedulerState)
	at io.enmasse.queue.scheduler.QueueScheduler.lambda$addressesChanged$9(QueueScheduler.java:139)
	at io.enmasse.queue.scheduler.QueueScheduler$$Lambda$44/1584530467.run(Unknown Source)
	at io.enmasse.queue.scheduler.QueueScheduler.lambda$executeBlocking$10(QueueScheduler.java:145)
	at io.enmasse.queue.scheduler.QueueScheduler$$Lambda$45/1607717068.handle(Unknown Source)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:278)
	at io.vertx.core.impl.ContextImpl$$Lambda$47/1554489389.run(Unknown Source)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:80)
	at io.vertx.core.impl.TaskQueue$$Lambda$9/1285524499.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
   Locked ownable synchronizers:
	- <0x00000000fc6859a0> (a java.util.concurrent.ThreadPoolExecutor$Worker)
````

Log file is here:
[queue_scheduler_excerpt.log](https://github.com/EnMasseProject/enmasse/files/1646403/queue_scheduler_excerpt.log)
Thread dump is here: 
[queue_scheduler_thread_dump.txt](https://github.com/EnMasseProject/enmasse/files/1646412/queue_scheduler_thread_dump.txt)

**Comment by lulf** [comment]
What is the pattern of creation here? Do you create all addresses at the same time, or do you add them one by one? Also  do you add/remove addresses quickly? The warnings _should_ be harmless, but we've seen the same logs in the CI. In that particular case, the test is frequently adding and removing addresses before the address has been fully configured. Also this log entry can occur if it is the last address on the broker, in which case the broker will be shutting down at the same time. We have #693 as well which causes a wrong status to be set for the address (ready even if it is not).

Do you have the corresponding broker log as well? 

Just FYI we are likely to be reworking the queue-scheduler with the goal of removing it in favor of a more robust approach over the next release(s).

**Comment by grs** [comment]
@lulf harmless because it should then recover (looks like its been blocked for nearly 4 minutes which seems very long)? Do we know *why* the thread is blocking? Is there a quick fix to give up on whatever it is blocking on after some timeout?

**Comment by lulf** [comment]
@grs 

The poll() done in the sendMessage is set to timeout after 30 seconds. The problem with the 4 minutes, is I believe that multiple address changes stacks  up due to the lock taken on the SchedulerState object while it is waiting for the response. If you have 8 of these changes in a row, and they all time out, the last change will easily be blocked for 4 minutes.

Question is why it is blocked and perhaps we should lower the timeout also, which is why i'd like to see the broker log, as the operations shouldn't really take long at all if everything is working fine.

It would be better if the logic would not hold the lock while querying the broker, but it was intended to ensure that two address changes would not interfere with each other.

**Comment by calohmn** [comment]
@lulf @grs 
It's 40 minutes in the log output above actually, or longer even if you look at the log timestamps:
````
Jan 19, 2018 9:54:42 AM io.vertx.core.impl.BlockedThreadChecker
WARNING: Thread Thread[vert.x-worker-thread-8,5,main] has been blocked for 60823 ms, time limit is 60000

...
Jan 19, 2018 11:03:22 AM io.vertx.core.impl.BlockedThreadChecker
WARNING: Thread Thread[vert.x-worker-thread-8,5,main] has been blocked for 2371642 ms, time limit is 60000
````

I had a look at the broker log and found nothing suspicious there - unfortunately I didn't save the log before deleting the pods to do a new test with 0.16.0-rc1 (couldn't reproduce the behaviour again yet).

Regarding the pattern of creation: the 13 addresses get created one after the other, without waiting for the "isReady" flag after each HTTP request.

**Comment by lulf** [comment]
@calohmn Ok, thanks. I will try to reproduce it.

**Comment by calohmn** [comment]
I got the error again and have attached some logs including the ones from the broker. I've enabled debug logging there.
[logs_180122.zip](https://github.com/EnMasseProject/enmasse/files/1650999/logs_180122.zip)

(Note that the start of the Artemis container is quite a bit delayed here because the docker image first couldn't be pulled and I had to rebuild it (based on the rc1 branch with added debugging).)

**Comment by lulf** [comment]
Thanks @calohmn 

From the queue-scheduler log I can count 83 address changes log messages (i.e. every change of an address status will cause a change here). This is far more than needed. The difference between this version and the previous versions are that configserv is no longer user, which provides some level of aggregation of these resource changes. So I think another change to the queue-scheduler should be to not distribute queues on every address change, but on a periodic base instead to reduce the 'unneeded' blocking for every one of those changes.

Looking more at the log, I see that the initial broker 'pooled-persisted-6dd664fcd8-4wvsc' is disconnected at 06:49:52. Then shortly afterwards a new broker connects:

06:49:54 INFO  QueueScheduler:119 - Connection opened from pooled-persisted-6dd664fcd8-xmxzl

The queues are created immediately. but only 13 seconds after, there are messages about them being destroyed. Unfortunately there are not log message saying what address state it is trying to update to, so I'm adding that.

@calohmn I'll build a snapshot for you with some fixes.

**Comment by calohmn** [comment]
@lulf One thing I find striking in the queue-scheduler log is that all addresses/queues get created twice:
First, there are the update events causing the queues to be created (in `io.enmasse.queue.scheduler.SchedulerState#groupUpdated()`) - all while the addresses still have "isReady: false". After that, creation of queues seems to be finished and no further action seems needed.

But then there is another update event coming with all contained addresses now having "isReady:false" and a different version id. This change of address version causes all queues to be deleted and recreated again.

I wonder if it would make sense to try to omit this queue-recreation happening along with the status change.

During one test run I encountered TimeoutExceptions during the recreation of queues:
````
2018-01-22 09:50:33 INFO  SchedulerState:74 - Updated addresses for pooled-persisted: [{address=event/HEALTH_CHECK_TENANT,name=event/HEALTH_CHECK_TENANT,uuid=92911f84-0e59-37e5-a9f4-e1b2f85b2d08,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=false,version=700908}, {address=event/DEFAULT_TENANT,name=event/DEFAULT_TENANT,uuid=96ea5893-7503-362d-ac96-048dbe8e183d,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=false,version=700904}, {address=event/SYSTEM_TEST_TENANT_2,name=event/SYSTEM_TEST_TENANT_2,uuid=99d326a6-9e80-38ea-a0c3-591a2f17a632,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=false,version=700916}, {address=event/DEVELOPER_UI_TENANT,name=event/DEVELOPER_UI_TENANT,uuid=e732c77a-c593-3be6-957b-e2237543bb54,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=false,version=700905}, {address=event/SYSTEM_TEST_TENANT_1,name=event/SYSTEM_TEST_TENANT_1,uuid=a6d54e7f-116d-342e-a817-6f81e3083f00,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=false,version=700915}, {address=event/PERF_TEST_TENANT,name=event/PERF_TEST_TENANT,uuid=c6b122bf-59b0-31fc-9f62-4c8fcb032899,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=false,version=700913}, {address=event/IOT_ACADEMY,name=event/IOT_ACADEMY,uuid=a9dbcfd6-11b3-3396-9794-dcc59527cbd5,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=false,version=700910}, {address=event/HUB_TEAM_TENANT,name=event/HUB_TEAM_TENANT,uuid=51ff3d5e-9590-33bb-8cd7-662f13205fac,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=false,version=700909}, {address=event/SMOKE_TEST_TENANT,name=event/SMOKE_TEST_TENANT,uuid=68dbefba-9efa-396c-ba0f-bc45cb701a98,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=false,version=700914}, {address=event/BCX18,name=event/BCX18,uuid=4ec7fe19-2e47-3edb-b19e-eb7efc6c9ce1,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=false,version=700903}, {address=event/LOAD_TEST_TENANT,name=event/LOAD_TEST_TENANT,uuid=2ed296c6-4409-3f57-955f-8588de7acb7e,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=false,version=700911}, {address=event/OCTO_TENANT,name=event/OCTO_TENANT,uuid=64cbd97d-18d8-33ab-ab42-fe0b4050ca39,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=false,version=700912}, {address=event/DX_TENANT,name=event/DX_TENANT,uuid=f144472d-5d8a-374d-adc1-76e4bc21ec06,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=false,version=700906}, {address=event/FREE_TENANT,name=event/FREE_TENANT,uuid=733d689e-ca16-39d2-b236-f13299ead089,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=false,version=700907}]
Jan 22, 2018 9:50:41 AM io.vertx.core.net.impl.NetServerImpl
SEVERE: Client from origin /172.17.0.1:53148 failed to connect over ssl: java.nio.channels.ClosedChannelException
2018-01-22 09:50:45 INFO  Artemis:218 - Destroyed connector service event/HEALTH_CHECK_TENANT
2018-01-22 09:50:45 INFO  Artemis:212 - Destroying queue event/HEALTH_CHECK_TENANT
2018-01-22 09:50:46 INFO  Artemis:218 - Destroyed connector service event/DEFAULT_TENANT
2018-01-22 09:50:46 INFO  Artemis:212 - Destroying queue event/DEFAULT_TENANT
2018-01-22 09:50:48 INFO  Artemis:218 - Destroyed connector service event/SYSTEM_TEST_TENANT_2
2018-01-22 09:50:48 INFO  Artemis:212 - Destroying queue event/SYSTEM_TEST_TENANT_2
Jan 22, 2018 9:50:51 AM io.vertx.core.net.impl.NetServerImpl
SEVERE: Client from origin /172.17.0.1:53362 failed to connect over ssl: java.nio.channels.ClosedChannelException
2018-01-22 09:50:53 INFO  Artemis:218 - Destroyed connector service event/DEVELOPER_UI_TENANT
2018-01-22 09:50:53 INFO  Artemis:212 - Destroying queue event/DEVELOPER_UI_TENANT
2018-01-22 09:50:53 INFO  Artemis:218 - Destroyed connector service event/SYSTEM_TEST_TENANT_1
2018-01-22 09:50:53 INFO  Artemis:212 - Destroying queue event/SYSTEM_TEST_TENANT_1
2018-01-22 09:50:54 INFO  Artemis:218 - Destroyed connector service event/PERF_TEST_TENANT
2018-01-22 09:50:54 INFO  Artemis:212 - Destroying queue event/PERF_TEST_TENANT
2018-01-22 09:50:55 INFO  Artemis:218 - Destroyed connector service event/IOT_ACADEMY
2018-01-22 09:50:55 INFO  Artemis:212 - Destroying queue event/IOT_ACADEMY
2018-01-22 09:50:55 INFO  Artemis:218 - Destroyed connector service event/HUB_TEAM_TENANT
2018-01-22 09:50:55 INFO  Artemis:212 - Destroying queue event/HUB_TEAM_TENANT
2018-01-22 09:50:55 INFO  Artemis:218 - Destroyed connector service event/SMOKE_TEST_TENANT
2018-01-22 09:50:55 INFO  Artemis:212 - Destroying queue event/SMOKE_TEST_TENANT
2018-01-22 09:50:55 INFO  Artemis:218 - Destroyed connector service event/BCX18
2018-01-22 09:50:55 INFO  Artemis:212 - Destroying queue event/BCX18
2018-01-22 09:50:55 INFO  Artemis:218 - Destroyed connector service event/LOAD_TEST_TENANT
2018-01-22 09:50:55 INFO  Artemis:212 - Destroying queue event/LOAD_TEST_TENANT
2018-01-22 09:50:55 INFO  Artemis:218 - Destroyed connector service event/OCTO_TENANT
2018-01-22 09:50:55 INFO  Artemis:212 - Destroying queue event/OCTO_TENANT
2018-01-22 09:50:55 INFO  Artemis:218 - Destroyed connector service event/DX_TENANT
2018-01-22 09:50:55 INFO  Artemis:212 - Destroying queue event/DX_TENANT
2018-01-22 09:50:55 INFO  Artemis:218 - Destroyed connector service event/FREE_TENANT
2018-01-22 09:50:55 INFO  Artemis:212 - Destroying queue event/FREE_TENANT
2018-01-22 09:50:55 INFO  Artemis:249 - Retrieving queue names
2018-01-22 09:50:55 INFO  Artemis:201 - Creating queue event/DEFAULT_TENANT with address event/DEFAULT_TENANT
2018-01-22 09:50:55 INFO  Artemis:206 - Creating connector service event/DEFAULT_TENANT
2018-01-22 09:50:55 INFO  Artemis:201 - Creating queue event/SMOKE_TEST_TENANT with address event/SMOKE_TEST_TENANT
2018-01-22 09:50:55 INFO  Artemis:206 - Creating connector service event/SMOKE_TEST_TENANT
2018-01-22 09:50:55 INFO  Artemis:201 - Creating queue event/OCTO_TENANT with address event/OCTO_TENANT
2018-01-22 09:50:55 INFO  Artemis:206 - Creating connector service event/OCTO_TENANT
2018-01-22 09:50:55 INFO  Artemis:201 - Creating queue event/FREE_TENANT with address event/FREE_TENANT
2018-01-22 09:50:55 INFO  Artemis:206 - Creating connector service event/FREE_TENANT
2018-01-22 09:50:56 INFO  Artemis:201 - Creating queue event/SYSTEM_TEST_TENANT_1 with address event/SYSTEM_TEST_TENANT_1
2018-01-22 09:50:56 INFO  Artemis:206 - Creating connector service event/SYSTEM_TEST_TENANT_1
2018-01-22 09:50:56 INFO  Artemis:201 - Creating queue event/SYSTEM_TEST_TENANT_2 with address event/SYSTEM_TEST_TENANT_2
2018-01-22 09:50:56 INFO  Artemis:206 - Creating connector service event/SYSTEM_TEST_TENANT_2
2018-01-22 09:50:56 INFO  Artemis:201 - Creating queue event/DEVELOPER_UI_TENANT with address event/DEVELOPER_UI_TENANT
2018-01-22 09:50:56 INFO  Artemis:206 - Creating connector service event/DEVELOPER_UI_TENANT
2018-01-22 09:50:56 INFO  Artemis:201 - Creating queue event/PERF_TEST_TENANT with address event/PERF_TEST_TENANT
2018-01-22 09:50:56 INFO  Artemis:206 - Creating connector service event/PERF_TEST_TENANT
2018-01-22 09:50:56 INFO  Artemis:201 - Creating queue event/BCX18 with address event/BCX18
2018-01-22 09:50:56 INFO  Artemis:206 - Creating connector service event/BCX18
2018-01-22 09:50:56 INFO  Artemis:201 - Creating queue event/HEALTH_CHECK_TENANT with address event/HEALTH_CHECK_TENANT
2018-01-22 09:50:56 INFO  Artemis:206 - Creating connector service event/HEALTH_CHECK_TENANT
2018-01-22 09:50:56 INFO  Artemis:201 - Creating queue event/IOT_ACADEMY with address event/IOT_ACADEMY
2018-01-22 09:50:56 INFO  Artemis:206 - Creating connector service event/IOT_ACADEMY
2018-01-22 09:50:56 INFO  Artemis:201 - Creating queue event/DX_TENANT with address event/DX_TENANT
Jan 22, 2018 9:51:01 AM io.vertx.core.net.impl.NetServerImpl
SEVERE: Client from origin /172.17.0.1:53630 failed to connect over ssl: java.nio.channels.ClosedChannelException
Jan 22, 2018 9:51:11 AM io.vertx.core.net.impl.NetServerImpl
SEVERE: Client from origin /172.17.0.1:53850 failed to connect over ssl: java.nio.channels.ClosedChannelException
Jan 22, 2018 9:51:21 AM io.vertx.core.net.impl.NetServerImpl
SEVERE: Client from origin /172.17.0.1:54072 failed to connect over ssl: java.nio.channels.ClosedChannelException
2018-01-22 09:51:26 ERROR QueueScheduler:153 - Error handling address change
java.util.concurrent.TimeoutException: Timed out getting response from broker pooled-persisted-6dd664fcd8-6jkjd on broker.createQueue with parameters: [event/DX_TENANT, ANYCAST, event/DX_TENANT, null, true, -1, false, true]
	at io.enmasse.amqp.Artemis.doOperation(Artemis.java:131)
	at io.enmasse.amqp.Artemis.createQueue(Artemis.java:202)
	at io.enmasse.queue.scheduler.ArtemisAdapter.createQueue(ArtemisAdapter.java:44)
	at io.enmasse.queue.scheduler.SchedulerState.deployQueue(SchedulerState.java:101)
	at io.enmasse.queue.scheduler.SchedulerState.distributeAddressesByNumQueues(SchedulerState.java:172)
	at io.enmasse.queue.scheduler.SchedulerState.addAddresses(SchedulerState.java:133)
	at io.enmasse.queue.scheduler.SchedulerState.groupUpdated(SchedulerState.java:70)
	at io.enmasse.queue.scheduler.SchedulerState.addressesChanged(SchedulerState.java:51)
	at io.enmasse.queue.scheduler.QueueScheduler.lambda$addressesChanged$9(QueueScheduler.java:140)
	at io.enmasse.queue.scheduler.QueueScheduler.lambda$executeBlocking$10(QueueScheduler.java:146)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:278)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:80)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Jan 22, 2018 9:51:31 AM io.vertx.core.net.impl.NetServerImpl
SEVERE: Client from origin /172.17.0.1:54224 failed to connect over ssl: java.nio.channels.ClosedChannelException
Jan 22, 2018 9:51:41 AM io.vertx.core.net.impl.NetServerImpl
SEVERE: Client from origin /172.17.0.1:54330 failed to connect over ssl: java.nio.channels.ClosedChannelException
Jan 22, 2018 9:51:51 AM io.vertx.core.net.impl.NetServerImpl
SEVERE: Client from origin /172.17.0.1:54438 failed to connect over ssl: java.nio.channels.ClosedChannelException
2018-01-22 09:51:56 ERROR QueueScheduler:153 - Error handling address change
java.util.concurrent.TimeoutException: Timed out getting response from broker pooled-persisted-6dd664fcd8-6jkjd on broker.destroyConnectorService with parameters: [event/HEALTH_CHECK_TENANT]
	at io.enmasse.amqp.Artemis.doOperation(Artemis.java:131)
	at io.enmasse.amqp.Artemis.destroyConnectorService(Artemis.java:217)
	at io.enmasse.queue.scheduler.ArtemisAdapter.deleteQueue(ArtemisAdapter.java:55)
	at io.enmasse.queue.scheduler.SchedulerState.deleteQueue(SchedulerState.java:105)
	at io.enmasse.queue.scheduler.SchedulerState.deleteAddresses(SchedulerState.java:201)
	at io.enmasse.queue.scheduler.SchedulerState.groupUpdated(SchedulerState.java:64)
	at io.enmasse.queue.scheduler.SchedulerState.addressesChanged(SchedulerState.java:51)
	at io.enmasse.queue.scheduler.QueueScheduler.lambda$addressesChanged$9(QueueScheduler.java:140)
	at io.enmasse.queue.scheduler.QueueScheduler.lambda$executeBlocking$10(QueueScheduler.java:146)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:278)
	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:80)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Jan 22, 2018 9:51:59 AM io.vertx.core.impl.BlockedThreadChecker
````

**Comment by lulf** [comment]
@calohmn Please try the latest snapshot: https://dl.bintray.com/enmasse/snapshots/enmasse-latest.tgz

It contains a fix for the issue describe in your latest comment as well.

**Comment by calohmn** [comment]
Thanks! With the above snapshot it looks good now.

**Comment by calohmn** [comment]
Just one minor thing:
Logging in the `SchedulerState.groupUpdated` method seems quite verbose now on level INFO.
Roughly every 10s I get

> 2018-01-23 16:23:18 INFO  SchedulerState:59 - Updating addresses for pooled-persisted: [{address=event/DX_TENANT,name=event/DX_TENANT,uuid=f144472d-5d8a-374d-adc1-76e4bc21ec06,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=true,version=756712}, {address=event/IOT_ACADEMY,name=event/IOT_ACADEMY,uuid=a9dbcfd6-11b3-3396-9794-dcc59527cbd5,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=true,version=756716}, {address=event/LOAD_TEST_TENANT,name=event/LOAD_TEST_TENANT,uuid=2ed296c6-4409-3f57-955f-8588de7acb7e,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=true,version=756717}, {address=event/SMOKE_TEST_TENANT,name=event/SMOKE_TEST_TENANT,uuid=68dbefba-9efa-396c-ba0f-bc45cb701a98,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=true,version=756720}, {address=event/FREE_TENANT,name=event/FREE_TENANT,uuid=733d689e-ca16-39d2-b236-f13299ead089,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=true,version=756713}, {address=event/HUB_TEAM_TENANT,name=event/HUB_TEAM_TENANT,uuid=51ff3d5e-9590-33bb-8cd7-662f13205fac,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=true,version=756715}, {address=event/DEVELOPER_UI_TENANT,name=event/DEVELOPER_UI_TENANT,uuid=e732c77a-c593-3be6-957b-e2237543bb54,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=true,version=756711}, {address=event/BCX18,name=event/BCX18,uuid=4ec7fe19-2e47-3edb-b19e-eb7efc6c9ce1,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=true,version=756708}, {address=event/HEALTH_CHECK_TENANT,name=event/HEALTH_CHECK_TENANT,uuid=92911f84-0e59-37e5-a9f4-e1b2f85b2d08,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=true,version=756714}, {address=event/SYSTEM_TEST_TENANT_1,name=event/SYSTEM_TEST_TENANT_1,uuid=a6d54e7f-116d-342e-a817-6f81e3083f00,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=true,version=756721}, {address=event/DEFAULT_TENANT,name=event/DEFAULT_TENANT,uuid=96ea5893-7503-362d-ac96-048dbe8e183d,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=true,version=756709}, {address=event/PERF_TEST_TENANT,name=event/PERF_TEST_TENANT,uuid=c6b122bf-59b0-31fc-9f62-4c8fcb032899,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=true,version=756719}, {address=event/OCTO_TENANT,name=event/OCTO_TENANT,uuid=64cbd97d-18d8-33ab-ab42-fe0b4050ca39,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=true,version=756718}, {address=event/SYSTEM_TEST_TENANT_2,name=event/SYSTEM_TEST_TENANT_2,uuid=99d326a6-9e80-38ea-a0c3-591a2f17a632,type=io.enmasse.address.model.AddressType@66f1911,plan=io.enmasse.address.model.Plan@8a42e3e1,status=true,version=756722}]
> 2018-01-23 16:23:18 INFO  SchedulerState:75 - Updated addresses for pooled-persisted

This is with 13 addresses. Imagining 100 addresses here would produce quite a log of clutter in the log.
Better to just log addresses that got actually added or deleted.

**Comment by lulf** [comment]
@calohmn Fixed as part of #787
